release-notes-1.2.0.md

```diff
@@ -0,0 +1,761 @@
+
+```ballerina
+xml x = xml `<root>
+<rectangle length="5" width="10"/>
+<rectangle length="5" width="5"/>
+<circle radius="2"/>
+<square side="6"/>
+</root>`;
+
+xml rectangles = x/*.<rectangle>;
+```
+
+#### XML attribute access
+
+ ```ballerina
+xmlns "www.ballerina.io/ns" as ns;
+xml val = xml `<element type="fixed" ns:count="2></element>`;
+string|error 'type = val.'type;
+string|error count = val.ns:count;
+```
+
+XML attribute access is now lax typed. This means that compile-time type checking is relaxed and moved to runtime. Accessing a non-existent attribute or using field access expression on a non-XML element item will result in an error being returned. If optional field access syntax is used, then nil will be returned instead of an error when the field is not available. Attributes with namespace prefix are accessed using the `value.prefix:attrName` syntax where the XML prefix must be declared in the scope.
+
+```ballerina
+xml val = xml `<element type="fixed"></element>`;
+string|error nonEx = val.nonExsistent; // result in error
+string|error? nonExOptional = val?.nonExsistent; // result in nil
+```
+
+### New language-integrated query design
+
+#### Query expression
+
+A query expression provides a language-integrated query feature using SQL-like syntax. 
+
+In its most basic form, a query expression consists of four kinds of clauses: from, let, where, and select. The first clause must be a from clause and the last clause must be a select. The result of the query expression is a list. In this basic form, a query expression is just a list comprehension.
+
+The from clause works similarly to a foreach statement. It creates an iterator from an iterable value and then binds variables to each value returned by the iterator. The where clause is a boolean expression, which can refer to variables bound by the from clause; when the where expression evaluates to false, the iteration skips following clauses. The let clause binds variables. The select clause is evaluated for each iteration; the result of the query expression is a list whose members are the result of the select clause. 
+
+```ballerina
+Person[] outputPersonList =
+       from var person in personList
+       let string depName = "WSO2", string replaceName = "Alexander"
+       where person.deptAccess == "A" && person.firstName == "Alex"
+       select {
+              firstName: replaceName,
+              lastName: person.lastName,
+              deptAccess: depName
+       };
+```
+
+As of now, query expressions are supported by lists and streams.
+
+#### Query action
+
+The clauses in the query pipeline of a query action are executed in the same way as the clauses in the query pipeline of a query expression. The query action is executed as follows. For each input frame `f` emitted by the query pipeline, execute the block-statement with `f` in the scope.
+
+```ballerina
+error? result = from var student in studentList
+ 	where student.score > 1.0
+do {
```

Comment:
Not formatted correctly.

Reply:
@hasithaa - Can you edit here itself so that I can commit directly?